File: backend/internal/core/domain/address.go
package domain

import "time"

// Address = ที่อยู่จัดส่งของผู้ใช้ — struct ล้วน ไม่มี tag ของ gorm
type Address struct {
	ID            uint
	UserID        uint
	RecipientName string
	Phone         string
	AddressLine   string
	Subdistrict   string
	District      string
	Province      string
	PostalCode    string
	Country       string
	Note          string
	IsDefault     bool
	CreatedAt     time.Time
	UpdatedAt     time.Time
}

File: frontend/src/types/address.ts
// สะท้อน response ของ /addresses ใน docs/rest_api.md (§7)

export interface Address {
  id: number
  recipient_name: string
  phone: string
  address_line: string
  subdistrict: string
  district: string
  province: string
  postal_code: string
  note?: string
  is_default: boolean
}

export interface AddressInput {
  recipient_name: string
  phone: string
  address_line: string
  subdistrict: string
  district: string
  province: string
  postal_code: string
  note?: string
  is_default?: boolean
}
